File: server/models/Project.js
module.exports = (sequelize, DataTypes) => {
  const Project = sequelize.define('Project', {
    projectId: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4, 
      primaryKey: true,
      
    },
    projectName: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    projectDeadline: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    batchId: {
      type: DataTypes.UUID,
  
      allowNull: false,
      references: {
        model: 'Batches',
        key: 'batchId',
      },
      onUpdate: 'CASCADE',
      onDelete: 'CASCADE',
    },
    visibility: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
  }, {
    timestamps: true,
  });

  Project.associate =models=>{
    Project.belongsTo(models.Batch,{foreignKey:'batchId',onDelete:'CASCADE'})
  }
  return Project;
};

File: eps-server-go/models/project.go
package models

import (
	"time"
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Project struct {
	ProjectID       uuid.UUID `gorm:"type:uuid;primaryKey" json:"projectId"`
	ProjectName     string    `gorm:"not null" json:"projectName"`
	ProjectDeadline time.Time `json:"projectDeadline"`
	BatchID         uuid.UUID `gorm:"type:uuid" json:"batchId"`
	Visibility      bool      `gorm:"default:true" json:"visibility"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`

	Batch Batch `gorm:"foreignKey:BatchID;constraint:OnDelete:CASCADE" json:"batch,omitempty"`
}

func (p *Project) BeforeCreate(tx *gorm.DB) (err error) {
	p.ProjectID = uuid.New()
	return
}